app/src/main/java/com/heig/atmanager/tasks/TaskFeedAdapter.java

```diff
@@ -101,7 +102,13 @@ public void onClick(View view) {
         holder.checkButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                tasks.get(position).setDone(holder.checkButton.isChecked());
+                if(holder.checkButton.isChecked()) {
+                    tasks.get(position).setDoneDate(new Date());
+                    tasks.get(position).setDone(true);
+                } else {
+                    tasks.get(position).setDoneDate(null);
+                    tasks.get(position).setDone(false);
+                }
```

Comment:
Not sure why change this : 
´´´tasks.get(position).setDone(holder.checkButton.isChecked()); ´´´
into an if else. Are you sure new Date() creates a new date of the current time ? If yes then ok

Reply:
yes it does. I also didn't want to call the function 2 times that is all